projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -79,9 +79,18 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     protected readonly renderingStrategy: RenderingStrategy,
     protected readonly svgUtilService: SvgUtilService,
     protected readonly d3Utils: D3UtilService,
-    protected readonly domRenderer: Renderer2
+    protected readonly domRenderer: Renderer2,
+    protected readonly groupId?: string
   ) {}
 
+  public showCrosshair(locationData: MouseLocationData<TData, Series<TData> | Band<TData>>[]): void {
+    if (locationData.length > 0) {
+      const location = locationData[0].location;
+      const currentLocation = this.allSeriesData.flatMap(viz => viz.dataForLocation({ x: location.x, y: location.y }));
+      this.renderedAxes.forEach(axis => axis.onMouseMove(currentLocation));
```

Comment:
I have no idea why this marked pending considering I've submitted reviews, but I'm guessing it never got seen, so reposting it here: 
![image](https://user-images.githubusercontent.com/45047841/159377073-b08007e1-fcb1-4a1f-b8bb-9c33e5634e89.png)

Reply:
its not causing infinite loop, since data is emitted from onMouseMove method. 
<img width="653" alt="Screenshot 2022-03-22 at 10 43 24 AM" src="https://user-images.githubusercontent.com/86998924/159412795-4c846b44-3804-42f9-8463-7fddd5f64497.png">

Reply:
Ah sorry - my mistake here was that we're calling `axis.onMouseMove`, I read it as `this.onMouseMove` which is what would have been the problem.

So this gets to two more points
1. We should expose explicit methods e.g. drawCrosshair and hideCrosshair that are called by the existing onMouseMove and onMouseLeave in axis. The problem doesn't exist in the current axis impl, but easily could - we don't want to rely on the handler not doing things other than drawing crosshairs, we're breaking encapsulation.
2. How is leave handled? When does the crosshair on the sync'd charts get hidden?

Reply:
2: passed empty data while moving mouse out and then if empty then this.renderedAxes.forEach(axis => axis.onMouseLeave()); still marker is there

Reply:
1 will work on this

Reply:
> 2: passed empty data on mouseleave and then if empty then this.renderedAxes.forEach(axis => axis.onMouseLeave()); still marker is there

Not sure I understand the point - this line of code is how we hide it for the current chart when the mouse leaves it - for the two that have been sync'd, how do they know to hide their tooltips?

Reply:
1) The marker was not removed on mouse leave for a single chart scenario. Reason class hidden was missing. when added a class named hidden marker is removed on mouse leave.
2) For removing the markers in other charts when the mouse leaves the first chart(current), the emitted  null as locationdata . On receiving null in other charts the above line of functionality called to hide the marker

Reply:
> The marker was not removed on mouse leave for a single chart scenario. Reason - class hidden was missing. when added a class named hidden marker is removed on mouse leave.

Thanks, that's a bug. Maybe it used to have a css class previously, but there's a global style on the attribute `hidden` so let's use that instead please.

> For removing the markers in other charts when the mouse leaves the first chart(current), the emitted null as locationdata . On receiving null in other charts the above line of functionality called to hide the marker

I just pulled this and tried it out. I made the hidden changes:
```diff
diff --git a/projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis-crosshair.ts b/projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis-crosshair.ts
index 8bc91d39..6fbd047e 100644
--- a/projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis-crosshair.ts
+++ b/projects/observability/src/shared/components/cartesian/d3/axis/cartesian-axis-crosshair.ts
@@ -30,11 +30,11 @@ export class CartesianAxisCrosshair {
       position,
       locationData.map(data => data.location)
     );
-    crosshairSelection.classed('hidden', false);
+    crosshairSelection.attr('hidden', null);
   }
 
   public hide(axisElement: SVGGElement): void {
-    this.getCrosshairSelection(axisElement).classed('hidden', true);
+    this.getCrosshairSelection(axisElement).attr('hidden', true);
   }
 
   private initializeCrosshairIfMissing(axisElement: SVGGElement): void {
```

But I see the crosshair on the other two charts remain after I mouse off. Am I missing a step?

Reply:
@aaron-steinfeld updated the PR with the above changes and marker removal for all charts.